src/react-apps/applications/receipt/src/features/receipt/containers/Receipt.tsx

```diff
@@ -0,0 +1,204 @@
+
+function Receipt(props: WithStyles<typeof styles>) {
+
+  const [party, setParty] = React.useState<IParty>(null);
+  const [instance, setInstance] = React.useState<IInstance>(null);
+  const [organizations, setOrganizations] = React.useState(null);
+  const [application, setApplication] = React.useState<IApplication>(null);
+  const [user, setUser] = React.useState<IProfile>(null);
+  const [language, setLanguage] = React.useState(null);
+  const isPrint = useMediaQuery('print');
+
+  const fetchParty = async () => {
+    try {
+      const response = await Axios.get<IParty>(getPartyUrl());
+      setParty(response.data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const fetchApplication = async () => {
+    try {
+      const app = instance.appId.split('/')[1];
+      const response = await Axios.get<IApplication>(getApplicationMetadataUrl(instance.org, app));
+      setApplication(response.data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const fetchInstance = async () => {
+    try {
+      const response = await Axios.get<IInstance>(getInstanceMetadataUrl());
+      setInstance(response.data);
+    } catch (error)  {
+      console.error(error);
+    }
+  };
+
+  const fetchOrganizations = async () => {
+    try {
+      const response = await Axios.get(altinnOrganisationsUrl);
+      setOrganizations(response.data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const fetchUser = async () => {
+    try {
+      const response = await Axios.get<IProfile>(getUserUrl());
+      setUser(response.data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const fetchLanguage = async () => {
+    try {
+      const response = await Axios.get(languageUrl);
+      setLanguage(response.data);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const getAttachments = (): IAttachment[] => {
+    if (!instance) {
+      return [];
+    } else {
+      const attachments: IAttachment[] = [];
+      instance.data.forEach((dataElement: IData) => {
+        if (dataElement.elementType !== 'default') {
+          attachments.push({
+          name: dataElement.fileName,
+          url: dataElement.dataLinks.platform,
+          iconClass: 'reg reg-attachment' });
+        }
+      });
+      return attachments;
+    }
+  };
+
+  const getTitle = (): string => {
+    const applicationTitle = application ? application.title.nb : '';
+    return `${applicationTitle} ${getLanguageFromKey('receipt_platform.is_sent', language)}`;
+  };
+
+  const handleModalClose = () => {
+    window.location.href = returnUrlToMessagebox(window.location.origin);
+  };
+
+  const isLoading = (): boolean => {
+    return (!party || !instance || !organizations || !application || !language || !user);
+  };
+
+  React.useEffect(() => {
+    if (!application && instance) {
+      fetchApplication();
+    }
+  }, [instance]);
+
+  React.useEffect(() => {
+    fetchInstance();
```

Comment:
All functions are executed asynchronously. I don't know how many of them are actual network outside the app... could they been executed concurrently with Axios.all?